apps/nestjs-wallet/src/vc-api/exchanges/entities/transaction.entity.ts

```diff
@@ -75,6 +76,54 @@ export class TransactionEntity {
   @Column('simple-json')
   callback: CallbackConfiguration[];
 
+  private verifyVpRequestTypeDidAuth(presentation: VerifiablePresentation): void {
+    // https://w3c-ccg.github.io/vp-request-spec/#did-authentication-request
+    if (!presentation.holder) throw Error('Presentation holder is required for didAuth query');
+  }
+
+  private verifyVpRequestTypePresentationDefinition(
+    presentation: VerifiablePresentation,
+    credentialQuery: Array<{ presentationDefinition: IPresentationDefinition }>
+  ): void {
+    // https://identity.foundation/presentation-exchange/#presentation-definition
+    const vcSubjects = new Set([...presentation.verifiableCredential.map((vc) => vc.credentialSubject.id)]);
+    if (vcSubjects.size > 1) throw Error('Credential subjects are not the same');
+
+    const pex: PEX = new PEX();
+
+    credentialQuery.forEach(({ presentationDefinition }) => {
+      const { errors } = pex.evaluatePresentation(
+        presentationDefinition,
+        presentation as unknown as IPresentation
```

Comment:
Instead of this type conversion, could we see if we can have the `VerifiablePresentation` extend the `IPresentation` type from `PEX` please?

Reply:
I will try to find a better way for that

Reply:
I can't extend it, however i found solution to remove unknown casting. Problem is that `IPresentation` interface doesn't allow inline json-ld contexts. Here is another reason to host our contexts.